engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -158,6 +157,19 @@ namespace dmGameSystem
         info->m_RefCount++;
     }
 
+    static uint32_t CalcBufferVersion(const MeshComponent* component, BufferResource* br)
+    {
+        HashState32 state;
+        dmHashInit32(&state, false);
+
+        uint32_t version;
+        dmBuffer::GetContentVersion(br->m_Buffer, &version);
+        dmHashUpdateBuffer32(&state, &br->m_Buffer, sizeof(br->m_Buffer)); // The handle is not a pointer, but a version+index
+        dmHashUpdateBuffer32(&state, &version, sizeof(version));
+
+        return dmHashFinal32(&state);
+    }
```

Comment:
One of the issues was that the buffer content version starts at 0 and counts upwards, and many other buffers may have a similar version.

So detecting a changed buffer requires more than the content version.